tests/hazmat/backends/test_openssl.py

```diff
@@ -478,6 +480,41 @@ def test_unsupported_ec_keys(self):
             backend.create_x509_csr(object(), private_key, hashes.SHA1())
 
 
+class TestOpenSSLSignX509Certificate(object):
+    def test_requires_certificate_builder(self):
+        private_key = RSA_KEY_2048.private_key(backend)
+
+        with pytest.raises(TypeError):
+            backend.sign_x509_certificate(object(), private_key, DummyHash())
+
+    def test_checks_for_unsupported_extensions(self):
+        private_key = RSA_KEY_2048.private_key(backend)
+        builder = x509.CertificateBuilder().subject_name(x509.Name([
+            x509.NameAttribute(x509.OID_COUNTRY_NAME, u'US'),
+            x509.NameAttribute(x509.OID_STATE_OR_PROVINCE_NAME, u'Texas'),
+            x509.NameAttribute(x509.OID_LOCALITY_NAME, u'Austin'),
+            x509.NameAttribute(x509.OID_ORGANIZATION_NAME, u'PyCA'),
+            x509.NameAttribute(x509.OID_COMMON_NAME, u'cryptography.io'),
+        ])).public_key(
+            private_key.public_key()
+        ).serial_number(
+            777
+        ).not_valid_before(
+            datetime.datetime(1999, 1, 1)
+        ).not_valid_after(
+            datetime.datetime(2020, 1, 1)
+        )
+
+        builder._extensions.append(x509.Extension(
```

Comment:
I'm confused by this comment. The CSRBuilder checks for extensions in [the backend][] as well as [the front end][]. In order to get proper test coverage, this seemed to be the easiest way to write the test that it would not allow a random extension.

[the backend]: https://github.com/pyca/cryptography/blob/93b5e3efab0337c51c84c27208d9034b607f09e2/src/cryptography/hazmat/backends/openssl/backend.py#L1030..L1039

[the front end]: https://github.com/pyca/cryptography/blob/93b5e3efab0337c51c84c27208d9034b607f09e2/src/cryptography/x509.py#L1572..L1585